diff --git a/sources/Adapters/PSP/FileSystem/PSPFileSystem.cpp b/sources/Adapters/PSP/FileSystem/PSPFileSystem.cpp
index 195963f..f2ce810 100644
--- a/sources/Adapters/PSP/FileSystem/PSPFileSystem.cpp
+++ b/sources/Adapters/PSP/FileSystem/PSPFileSystem.cpp
@@ -4,7 +4,7 @@
 #include <string.h>
 #include <stdarg.h>
 #include <string>
-#include <Application/utils/wildcard.h>
+#include <Application/Utils/wildcard.h>
 
 #include <sys/dir.h>
 #include <sys/stat.h>
@@ -177,4 +177,4 @@ Result PSPFileSystem::MakeDir(const char *path) {
     return Result(oss.str());
   }
   return Result::NoError;
-}
\ No newline at end of file
+}
